$NetBSD: patch-ah,v 1.2 2010/05/03 02:10:43 jnemeth Exp $

* Call function_return() with IGNORE_THREAD after calling thread_release().

--- DriverManager/SQLTransact.c.orig	2010-04-13 16:56:08.000000000 +0000
+++ DriverManager/SQLTransact.c
@@ -465,7 +465,7 @@ SQLRETURN SQLTransact( SQLHENV environme
 
             thread_release( SQL_HANDLE_ENV, environment );
 
-            return function_return( SQL_HANDLE_ENV, environment, SQL_ERROR );
+            return function_return( IGNORE_THREAD, environment, SQL_ERROR );
         }
 
         if ( environment -> state == STATE_E2 )
@@ -536,7 +536,7 @@ SQLRETURN SQLTransact( SQLHENV environme
 
                             thread_release( SQL_HANDLE_ENV, environment );
 
-                            return function_return( SQL_HANDLE_ENV, environment, SQL_ERROR );
+                            return function_return( IGNORE_THREAD, environment, SQL_ERROR );
                         }
                     }
                     else if ( CHECK_SQLENDTRAN( connection ))
